src/lsqecc/gates/gates.py

```diff
@@ -49,26 +55,59 @@ class T(Gate):
     pass
 
 
+@dataclass
+class PauliRotations(Gate):
+    phase: Fraction = Fraction(1, 1)
+    axis: PauliOperator = PauliOperator.Z
+
+    def to_clifford_plus_t(self, compress_rotations: bool = False) -> Sequence["Gate"]:
+        if self.axis == PauliOperator.Z:
+            if self.phase == Fraction(1, 1):
+                return [Z(self.target_qubit)]
+            elif self.phase == Fraction(1, 2):
+                return [S(self.target_qubit)]
+            elif self.phase == Fraction(1, 4):
+                return [T(self.target_qubit)]
+            else:
+                return approximate.approximate_rz(RZ(phase=self.phase), compress_rotations)
```

Comment:
I checked the logic for this and it was not being called! Did you find it otherwise ?

Reply:
:+1:

Reply:
No I haven't check, guess we would find out if rotations expand again. In general though I might replace this return with an `assert False`, just so that we avoid chasing down complicated logic.